kotlin/src/main/kotlin/org/jdbi/v3/core/kotlin/Jdbi858Extensions.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("unused")
+
+package org.jdbi.v3.core.kotlin
+
+import org.jdbi.v3.core.Handle
+import org.jdbi.v3.core.HandleCallback
+import org.jdbi.v3.core.HandleConsumer
+import org.jdbi.v3.core.Jdbi
+import org.jdbi.v3.core.extension.ExtensionCallback
+import org.jdbi.v3.core.extension.ExtensionConsumer
+import org.jdbi.v3.core.transaction.TransactionIsolationLevel
+
+// The extensions in this file were created in response to these issues :
+// https://github.com/jdbi/jdbi/issues/858
+// https://youtrack.jetbrains.com/issue/KT-5464
+
+// These extensions are temporary and will be removed when Kotlin's type inference
+// is improved and this isn't an issue anymore.
+
+/**
+ * Temporary extension function for [Jdbi.withHandle].
+ *
+ * This function WILL be deprecated when not needed anymore.
```

Comment:
Let's state this more strongly: deprecated, and later removed.

Ditto for the other extension methods.